quesma/clickhouse/clickhouse.go

```diff
@@ -30,18 +30,31 @@ import (
 
 const (
 	timestampFieldName = "@timestamp" // it's always DateTime64 for now, don't want to waste time changing that, we don't seem to use that anyway
+	// Above this number of columns we will use heuristic
+	// to decide if we should add new columns
+	maxColumns = 100
+)
+
+type (
+	IngestFieldBucketKey struct {
+		indexName string
+		field     string
+	}
+	IngestFieldStatistics map[IngestFieldBucketKey]int64
```

Comment:
Not sure if it's thread safe, would take the lock.

Reply:
Yes. Good point